src/rashdf/geom.py

```diff
@@ -142,13 +142,105 @@ def mesh_cell_faces(self) -> GeoDataFrame:
         return GeoDataFrame(face_dict, geometry="geometry", crs=self.projection())
 
     def bc_lines(self) -> GeoDataFrame:
-        raise NotImplementedError
+        """Return the 2D mesh area boundary condition lines.
+        
+        Returns
+        -------
+        GeoDataFrame
+            A GeoDataFrame containing the 2D mesh area boundary condition lines if they exist.
+        """
+        if "/Geometry/Boundary Condition Lines" not in self:
+            return GeoDataFrame()
+        bc_line_data = self["/Geometry/Boundary Condition Lines"]
+        bc_line_ids = range(bc_line_data["Attributes"][()].shape[0])
+        v_conv_str = np.vectorize(convert_ras_hdf_string)
+        names = v_conv_str(bc_line_data["Attributes"][()]["Name"])
+        mesh_names = v_conv_str(bc_line_data["Attributes"][()]["SA-2D"])
+        types = v_conv_str(bc_line_data["Attributes"][()]["Type"])
+        multi_lines = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in bc_line_data["Polyline Info"][()]:
+            points = bc_line_data["Polyline Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = bc_line_data["Polyline Parts"][()][part_start:part_start+part_cnt]
+            multi_lines.append(
+                MultiLineString(
+                    list(points[part_pnt_start:part_pnt_start+part_pnt_cnt] for part_pnt_start, part_pnt_cnt in parts)
+                )
+            )
+        return GeoDataFrame(
+            {
+                "bc_line_id":bc_line_ids,
+                "name":names,
+                "mesh_name":mesh_names,
+                "type":types,
+                "geometry":multi_lines
+            },
+            geometry="geometry",
+            crs=self.projection()
+        )
 
     def breaklines(self) -> GeoDataFrame:
-        raise NotImplementedError
+        """Return the 2D mesh area breaklines.
+        
+        Returns
+        -------
+        GeoDataFrame
+            A GeoDataFrame containing the 2D mesh area breaklines if they exist.
+        """
+        if "/Geometry/2D Flow Area Break Lines" not in self:
+            return GeoDataFrame()
+        bl_line_data = self["/Geometry/2D Flow Area Break Lines"]
+        bl_line_ids = range(bl_line_data["Attributes"][()].shape[0])
+        names = np.vectorize(convert_ras_hdf_string)(bl_line_data["Attributes"][()]["Name"])
+        multi_lines = list()
+        for pnt_start, pnt_cnt, part_start, part_cnt in bl_line_data["Polyline Info"][()]:
+            points = bl_line_data["Polyline Points"][()][pnt_start:pnt_start+pnt_cnt]
+            parts = bl_line_data["Polyline Parts"][()][part_start:part_start+part_cnt]
+            multi_lines.append(
+                MultiLineString(
```

Comment:
Similar to previous comment -- can we use `LineString` for single-part lines and `MultiLineString` for multi-part?

Reply:
Updated to use `LineString` when single-part.